deltacat/aws/clients.py

```diff
@@ -93,7 +93,15 @@ def retrying_get(
             after=_log_attempt_number,
         ):
             with attempt:
-                resp = _get_url(url)
+                try:
+                    resp = _get_url(url)
+                except requests.exceptions.HTTPError as exception:
+                    # Unauthorized errors imply IMDSv2 is being used
+                    if exception.response.status_code == HTTPStatus.UNAUTHORIZED:
+                        return None
```

Comment:
I believe this would still be required.

Reply:
The idea here is to allowlist this case - let the ping exit successfully, instead of fail or retry numerous times. This exception only occurs if IMDSv2 is enabled. 

> What is your proposal for pinging IMDSv2?

We will already be pinging IMDSv2 if it is enabled on the EC2 instance. The main difference with IMDSv1 and IMDSv2 is that an additional call to fetch the session token is required, which will then be passed to subsequent EC2 metadata API calls. For pinging purposes, I avoided this step.